### Pandas Lab -- Cleaning, Merging, & Grouping

This lab is designed to introduce students to common use cases for Pandas when working with data:

 - Creating new information out of your existing data set
 - Merging, concatenating, and joining different data sources
 - Grouping -- With both time & non-time based data

### Section I: Creating Data Out of Your Existing Columns

Go ahead and create the following columns in your dataset.

**Column 1:**

  - **Column Name:** Weekend
  - **Values:** `True` if `day_of_week` is either Friday or Saturday, `False` if not

In [127]:
import pandas as pd
import numpy as np
# read in the file
df = pd.read_csv('/Users/mcs275/dat-11-15/ClassMaterial/Unit1/data/master.csv', parse_dates=['visit_date'])

In [128]:
df.columns

Index(['id', 'visit_date', 'visitors', 'day_of_week', 'holiday', 'genre',
       'area', 'latitude', 'longitude', 'reserve_visitors'],
      dtype='object')

In [2]:
df['Weekend'] = df['day_of_week'].isin(["Friday", "Saturday"])

In [129]:
df

id visit_date  visitors day_of_week  holiday  \
0       air_ba937bf13d40fb24 2016-01-13        25   Wednesday        0   
1       air_ba937bf13d40fb24 2016-01-14        32    Thursday        0   
2       air_ba937bf13d40fb24 2016-01-15        29      Friday        0   
3       air_ba937bf13d40fb24 2016-01-16        22    Saturday        0   
4       air_ba937bf13d40fb24 2016-01-18         6      Monday        0   
...                      ...        ...       ...         ...      ...   
252103  air_a17f0778617c76e2 2017-04-21        49      Friday        0   
252104  air_a17f0778617c76e2 2017-04-22        60    Saturday        0   
252105  air_a17f0778617c76e2 2017-03-26        69      Sunday        0   
252106  air_a17f0778617c76e2 2017-03-20        31      Monday        1   
252107  air_a17f0778617c76e2 2017-04-09        26      Sunday        0   

                 genre                          area   latitude   longitude  \
0           Dining bar  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599   
1           Dining bar  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599   
2           Dining bar  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599   
3           Dining bar  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599   
4           Dining bar  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599   
...                ...                           ...        ...         ...   
252103  Italian/French  Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197852   
252104  Italian/French  Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197852   
252105  Italian/French  Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197852   
252106  Italian/French  Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197852   
252107  Italian/French  Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197852   

        reserve_visitors  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  
...                  ...  
252103               6.0  
252104              37.0  
252105              35.0  
252106               3.0  
252107              32.0  

[252108 rows x 10 columns]

**Column 2:**

 - **Column Name:** Reservation Activity
 - **Values:**
   - `Low` if `reserve_visitors` is in the bottom .25 percentile
   - `Medium` if `reserve_visitors` is in the middle .50 percentile
   - `High`if `reserve_visitors` is in the top .25 percentile
   
**Hint:** Use the `quantile` method to get this value

In [4]:
conditions = [(df['reserve_visitors'] <= 4.0),
              ((df['reserve_visitors'] > 4.0) & (df['reserve_visitors'] < 24)),
              (df['reserve_visitors'] >= 24)
             ]
results    = [
    'Low',
    'Medium',
    'High'
]
df['Reservation_Activity'] = np.select(conditions, results, 'N/A')

In [ ]:
##using quantile in condidtions
conditions = [df['reserve_visitors'] < df['reserve_visitors'].quantile(0.25),
              df['reserve_visitors'].between(df['reserve_visitors'].quantile(0.25), df['reserve_visitors'].quantile(0.75)),
              df['reserve_visitors'] > df['reserve_visitors'].quantile(0.75)]

In [5]:
df.head()

id visit_date  visitors day_of_week  holiday       genre  \
0  air_ba937bf13d40fb24 2016-01-13        25   Wednesday        0  Dining bar   
1  air_ba937bf13d40fb24 2016-01-14        32    Thursday        0  Dining bar   
2  air_ba937bf13d40fb24 2016-01-15        29      Friday        0  Dining bar   
3  air_ba937bf13d40fb24 2016-01-16        22    Saturday        0  Dining bar   
4  air_ba937bf13d40fb24 2016-01-18         6      Monday        0  Dining bar   

                           area   latitude   longitude  reserve_visitors  \
0  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
1  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
2  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
3  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
4  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   

   Weekend Reservation_Activity  
0    False                  N/A  
1    False                  N/A  
2     True                  N/A  
3     True                  N/A  
4    False                  N/A

**Column 3:**

 - **Column Name:** Days
 - **Values:**
   - The length of time that has passed from the beginning of the time series, in days
 - **Note:** When you subtract these columns, your column will be a **time delta**.  See if you can use the `dt` attribute to convert these values into an integer.  Ie, if your value reads `3 days`, you want that to be 3 instead.  You can read more about different time periods in pandas here:  https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#time-date-components

In [6]:
df['days'] = df['visit_date'] - df['visit_date'].min()
df['days'] = df['days'].dt.days

In [7]:
df.head()

id visit_date  visitors day_of_week  holiday       genre  \
0  air_ba937bf13d40fb24 2016-01-13        25   Wednesday        0  Dining bar   
1  air_ba937bf13d40fb24 2016-01-14        32    Thursday        0  Dining bar   
2  air_ba937bf13d40fb24 2016-01-15        29      Friday        0  Dining bar   
3  air_ba937bf13d40fb24 2016-01-16        22    Saturday        0  Dining bar   
4  air_ba937bf13d40fb24 2016-01-18         6      Monday        0  Dining bar   

                           area   latitude   longitude  reserve_visitors  \
0  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
1  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
2  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
3  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   
4  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599               NaN   

   Weekend Reservation_Activity  days  
0    False                  N/A    12  
1    False                  N/A    13  
2     True                  N/A    14  
3     True                  N/A    15  
4    False                  N/A    17

### Section II: Merging Dataframes

The dataset we have been working with so far (`master.csv`), is actually a combined version of several datasets.  

In this section of the lab, we are going to re-create it manually from its individual pieces.

In the `restaurant data` folder, you'll find the following files:

 - `air_reserve.csv`
 - `air_store_info.csv`
 - `air_visit_data.csv`
 - `date_info.csv`
 
They contain all the constituent info for the `master.csv` file that we're currently using. 

You should have 252108 rows when you are finished.

Using merges, piece the files together to recreate the one we are currently working on.  

**Hint:** To get the number of reservations in the `reserve_visitors` column, you will have to use the `groupby` method first for each `store_id` and `day` before doing the merging.

You will also have to make sure each column is the same datatype -- date is probably the best.  Check the amount of null values in the new column to ensure you did it correctly.  (An incorrect merge will have no non-null values).

Some operations that might come in handy:

 - `dt.date` -- converts a datetime to a date
 - `pd.to_datetime` if you need to convert something from a string to a date

In [66]:
reservations = pd.read_csv('/Users/mcs275/dat-class-repo/ClassMaterial/Unit1/data/restaurant_data/air_reserve.csv')

In [41]:
info = pd.read_csv('/Users/mcs275/dat-class-repo/ClassMaterial/Unit1/data/restaurant_data/air_store_info.csv')

In [42]:
visits = pd.read_csv('/Users/mcs275/dat-class-repo/ClassMaterial/Unit1/data/restaurant_data/air_visit_data.csv')

In [43]:
dates = pd.read_csv('/Users/mcs275/dat-class-repo/ClassMaterial/Unit1/data/restaurant_data/date_info.csv')

In [98]:
visits['visit_date'] = pd.to_datetime(visits['visit_date'])

In [132]:
info

air_store_id         air_genre_name  \
0    air_0f0cdeee6c9bf3d7         Italian/French   
1    air_7cc17a324ae5c7dc         Italian/French   
2    air_fee8dcf4d619598e         Italian/French   
3    air_a17f0778617c76e2         Italian/French   
4    air_83db5aff8f50478e         Italian/French   
..                    ...                    ...   
824  air_9bf595ef095572fb  International cuisine   
825  air_764f71040a413d4d                  Asian   
826  air_10bbe8acd943d8f6                  Asian   
827  air_7514d90009613cd6          Karaoke/Party   
828  air_c8fe396d6c46275d          Karaoke/Party   

                             air_area_name   latitude   longitude  
0             Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197853  
1             Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197853  
2             Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197853  
3             Hyōgo-ken Kōbe-shi Kumoidōri  34.695124  135.197853  
4             Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599  
..                                     ...        ...         ...  
824            Tōkyō-to Shibuya-ku Shibuya  35.661777  139.704051  
825            Tōkyō-to Shibuya-ku Shibuya  35.661777  139.704051  
826            Tōkyō-to Shibuya-ku Shibuya  35.661777  139.704051  
827  Hokkaidō Sapporo-shi Minami 3 Jōnishi  43.055460  141.340956  
828  Hokkaidō Sapporo-shi Minami 3 Jōnishi  43.055460  141.340956  

[829 rows x 5 columns]

In [44]:
reservations.head()

air_store_id       visit_datetime     reserve_datetime  \
0  air_877f79706adbfb06  2016-01-01 19:00:00  2016-01-01 16:00:00   
1  air_db4b38ebe7a7ceff  2016-01-01 19:00:00  2016-01-01 19:00:00   
2  air_db4b38ebe7a7ceff  2016-01-01 19:00:00  2016-01-01 19:00:00   
3  air_877f79706adbfb06  2016-01-01 20:00:00  2016-01-01 16:00:00   
4  air_db80363d35f10926  2016-01-01 20:00:00  2016-01-01 01:00:00   

   reserve_visitors  
0                 1  
1                 3  
2                 6  
3                 2  
4                 5

In [45]:
visits.head()

air_store_id  visit_date  visitors
0  air_ba937bf13d40fb24  2016-01-13        25
1  air_ba937bf13d40fb24  2016-01-14        32
2  air_ba937bf13d40fb24  2016-01-15        29
3  air_ba937bf13d40fb24  2016-01-16        22
4  air_ba937bf13d40fb24  2016-01-18         6

In [47]:
reservations.head()

air_store_id       visit_datetime     reserve_datetime  \
0  air_877f79706adbfb06  2016-01-01 19:00:00  2016-01-01 16:00:00   
1  air_db4b38ebe7a7ceff  2016-01-01 19:00:00  2016-01-01 19:00:00   
2  air_db4b38ebe7a7ceff  2016-01-01 19:00:00  2016-01-01 19:00:00   
3  air_877f79706adbfb06  2016-01-01 20:00:00  2016-01-01 16:00:00   
4  air_db80363d35f10926  2016-01-01 20:00:00  2016-01-01 01:00:00   

   reserve_visitors  
0                 1  
1                 3  
2                 6  
3                 2  
4                 5

In [99]:
df1 = visits.merge(info, on='air_store_id', how='left')

In [100]:
df1.head()

air_store_id visit_date  visitors air_genre_name  \
0  air_ba937bf13d40fb24 2016-01-13        25     Dining bar   
1  air_ba937bf13d40fb24 2016-01-14        32     Dining bar   
2  air_ba937bf13d40fb24 2016-01-15        29     Dining bar   
3  air_ba937bf13d40fb24 2016-01-16        22     Dining bar   
4  air_ba937bf13d40fb24 2016-01-18         6     Dining bar   

                  air_area_name   latitude   longitude  
0  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599  
1  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599  
2  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599  
3  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599  
4  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599

In [101]:
dates.head()

calendar_date day_of_week  holiday_flg
0    2016-01-01      Friday            1
1    2016-01-02    Saturday            1
2    2016-01-03      Sunday            1
3    2016-01-04      Monday            0
4    2016-01-05     Tuesday            0

In [104]:
dates['calendar_date'] = pd.to_datetime(dates['calendar_date'])

In [105]:
df1 = df1.merge(dates, left_on = 'visit_date', right_on = 'calendar_date', how = 'left')

In [106]:
df1.head()

air_store_id visit_date  visitors air_genre_name  \
0  air_ba937bf13d40fb24 2016-01-13        25     Dining bar   
1  air_ba937bf13d40fb24 2016-01-14        32     Dining bar   
2  air_ba937bf13d40fb24 2016-01-15        29     Dining bar   
3  air_ba937bf13d40fb24 2016-01-16        22     Dining bar   
4  air_ba937bf13d40fb24 2016-01-18         6     Dining bar   

                  air_area_name   latitude   longitude calendar_date  \
0  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-13   
1  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-14   
2  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-15   
3  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-16   
4  Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-18   

  day_of_week  holiday_flg  
0   Wednesday            0  
1    Thursday            0  
2      Friday            0  
3    Saturday            0  
4      Monday            0

In [107]:
reservations.columns

Index(['air_store_id', 'visit_datetime', 'reserve_datetime',
       'reserve_visitors', 'visit_date'],
      dtype='object')

In [108]:
reservations

air_store_id      visit_datetime     reserve_datetime  \
0      air_877f79706adbfb06 2016-01-01 19:00:00  2016-01-01 16:00:00   
1      air_db4b38ebe7a7ceff 2016-01-01 19:00:00  2016-01-01 19:00:00   
2      air_db4b38ebe7a7ceff 2016-01-01 19:00:00  2016-01-01 19:00:00   
3      air_877f79706adbfb06 2016-01-01 20:00:00  2016-01-01 16:00:00   
4      air_db80363d35f10926 2016-01-01 20:00:00  2016-01-01 01:00:00   
...                     ...                 ...                  ...   
92373  air_db4b38ebe7a7ceff 2017-05-29 18:00:00  2017-04-17 19:00:00   
92374  air_1033310359ceeac1 2017-05-30 23:00:00  2017-04-19 17:00:00   
92375  air_877f79706adbfb06 2017-05-31 20:00:00  2017-04-12 23:00:00   
92376  air_900d755ebd2f7bbd 2017-05-31 20:00:00  2017-04-19 16:00:00   
92377  air_3cad29d1a23209d2 2017-05-31 21:00:00  2017-04-22 10:00:00   

       reserve_visitors  visit_date  
0                     1  2016-01-01  
1                     3  2016-01-01  
2                     6  2016-01-01  
3                     2  2016-01-01  
4                     5  2016-01-01  
...                 ...         ...  
92373                 6  2017-05-29  
92374                 8  2017-05-30  
92375                 3  2017-05-31  
92376                10  2017-05-31  
92377                 3  2017-05-31  

[92378 rows x 5 columns]

In [109]:
reservations['visit_datetime'] = pd.to_datetime(reservations['visit_datetime'])

In [110]:
reservations['visit_date'] = reservations['visit_datetime'].dt.date

In [111]:
reservations.head()

air_store_id      visit_datetime     reserve_datetime  \
0  air_877f79706adbfb06 2016-01-01 19:00:00  2016-01-01 16:00:00   
1  air_db4b38ebe7a7ceff 2016-01-01 19:00:00  2016-01-01 19:00:00   
2  air_db4b38ebe7a7ceff 2016-01-01 19:00:00  2016-01-01 19:00:00   
3  air_877f79706adbfb06 2016-01-01 20:00:00  2016-01-01 16:00:00   
4  air_db80363d35f10926 2016-01-01 20:00:00  2016-01-01 01:00:00   

   reserve_visitors  visit_date  
0                 1  2016-01-01  
1                 3  2016-01-01  
2                 6  2016-01-01  
3                 2  2016-01-01  
4                 5  2016-01-01

In [112]:
reservations['visit_datetime'].dtype

dtype('<M8[ns]')

In [113]:
reservations['visit_date'].dtype

dtype('O')

In [114]:
reservations_grouped = reservations.groupby(['air_store_id','visit_date']).sum()

In [115]:
reservations_grouped = reservations_grouped.reset_index()

In [116]:
reservations_grouped.head()

air_store_id  visit_date  reserve_visitors
0  air_00a91d42b08b08d9  2016-10-31                 2
1  air_00a91d42b08b08d9  2016-12-05                 9
2  air_00a91d42b08b08d9  2016-12-14                18
3  air_00a91d42b08b08d9  2016-12-17                 2
4  air_00a91d42b08b08d9  2016-12-20                 4

In [117]:
df1['visit_date'] = pd.to_datetime(df1['visit_date'])

In [121]:
df1['visit_date'] = df1['visit_date'].dt.date

In [122]:
df2 = df1.merge(reservations_grouped, on = ['air_store_id','visit_date'], how='left')

In [123]:
df2['reserve_visitors'].max()

1633.0

In [125]:
df2.isnull()

air_store_id  visit_date  visitors  air_genre_name  air_area_name  \
0              False       False     False           False          False   
1              False       False     False           False          False   
2              False       False     False           False          False   
3              False       False     False           False          False   
4              False       False     False           False          False   
...              ...         ...       ...             ...            ...   
252103         False       False     False           False          False   
252104         False       False     False           False          False   
252105         False       False     False           False          False   
252106         False       False     False           False          False   
252107         False       False     False           False          False   

        latitude  longitude  calendar_date  day_of_week  holiday_flg  \
0          False      False          False        False        False   
1          False      False          False        False        False   
2          False      False          False        False        False   
3          False      False          False        False        False   
4          False      False          False        False        False   
...          ...        ...            ...          ...          ...   
252103     False      False          False        False        False   
252104     False      False          False        False        False   
252105     False      False          False        False        False   
252106     False      False          False        False        False   
252107     False      False          False        False        False   

        reserve_visitors  
0                   True  
1                   True  
2                   True  
3                   True  
4                   True  
...                  ...  
252103              True  
252104              True  
252105              True  
252106              True  
252107              True  

[252108 rows x 11 columns]

In [126]:
df2.columns

Index(['air_store_id', 'visit_date', 'visitors', 'air_genre_name',
       'air_area_name', 'latitude', 'longitude', 'calendar_date',
       'day_of_week', 'holiday_flg', 'reserve_visitors'],
      dtype='object')

In [130]:
df2['reserve_visitors'].min()

1.0

In [133]:
df2.head(50)

air_store_id  visit_date  visitors air_genre_name  \
0   air_ba937bf13d40fb24  2016-01-13        25     Dining bar   
1   air_ba937bf13d40fb24  2016-01-14        32     Dining bar   
2   air_ba937bf13d40fb24  2016-01-15        29     Dining bar   
3   air_ba937bf13d40fb24  2016-01-16        22     Dining bar   
4   air_ba937bf13d40fb24  2016-01-18         6     Dining bar   
5   air_ba937bf13d40fb24  2016-01-19         9     Dining bar   
6   air_ba937bf13d40fb24  2016-01-20        31     Dining bar   
7   air_ba937bf13d40fb24  2016-01-21        21     Dining bar   
8   air_ba937bf13d40fb24  2016-01-22        18     Dining bar   
9   air_ba937bf13d40fb24  2016-01-23        26     Dining bar   
10  air_ba937bf13d40fb24  2016-01-25        21     Dining bar   
11  air_ba937bf13d40fb24  2016-01-26        11     Dining bar   
12  air_ba937bf13d40fb24  2016-01-27        24     Dining bar   
13  air_ba937bf13d40fb24  2016-01-28        21     Dining bar   
14  air_ba937bf13d40fb24  2016-01-29        26     Dining bar   
15  air_ba937bf13d40fb24  2016-01-30         6     Dining bar   
16  air_ba937bf13d40fb24  2016-02-03        18     Dining bar   
17  air_ba937bf13d40fb24  2016-02-04        12     Dining bar   
18  air_ba937bf13d40fb24  2016-02-05        45     Dining bar   
19  air_ba937bf13d40fb24  2016-02-06        15     Dining bar   
20  air_ba937bf13d40fb24  2016-02-08        19     Dining bar   
21  air_ba937bf13d40fb24  2016-02-09        15     Dining bar   
22  air_ba937bf13d40fb24  2016-02-10        32     Dining bar   
23  air_ba937bf13d40fb24  2016-02-11         3     Dining bar   
24  air_ba937bf13d40fb24  2016-02-12        26     Dining bar   
25  air_ba937bf13d40fb24  2016-02-13         8     Dining bar   
26  air_ba937bf13d40fb24  2016-02-15        14     Dining bar   
27  air_ba937bf13d40fb24  2016-02-16        15     Dining bar   
28  air_ba937bf13d40fb24  2016-02-17        17     Dining bar   
29  air_ba937bf13d40fb24  2016-02-18        22     Dining bar   
30  air_ba937bf13d40fb24  2016-02-19        43     Dining bar   
31  air_ba937bf13d40fb24  2016-02-20        20     Dining bar   
32  air_ba937bf13d40fb24  2016-02-22         7     Dining bar   
33  air_ba937bf13d40fb24  2016-02-23        16     Dining bar   
34  air_ba937bf13d40fb24  2016-02-24        21     Dining bar   
35  air_ba937bf13d40fb24  2016-02-25        21     Dining bar   
36  air_ba937bf13d40fb24  2016-02-26        32     Dining bar   
37  air_ba937bf13d40fb24  2016-02-27        23     Dining bar   
38  air_ba937bf13d40fb24  2016-02-29        22     Dining bar   
39  air_ba937bf13d40fb24  2016-03-01        19     Dining bar   
40  air_ba937bf13d40fb24  2016-03-02        21     Dining bar   
41  air_ba937bf13d40fb24  2016-03-03        20     Dining bar   
42  air_ba937bf13d40fb24  2016-03-04        37     Dining bar   
43  air_ba937bf13d40fb24  2016-03-05        13     Dining bar   
44  air_ba937bf13d40fb24  2016-03-07        11     Dining bar   
45  air_ba937bf13d40fb24  2016-03-08         8     Dining bar   
46  air_ba937bf13d40fb24  2016-03-09        23     Dining bar   
47  air_ba937bf13d40fb24  2016-03-10        31     Dining bar   
48  air_ba937bf13d40fb24  2016-03-11        37     Dining bar   
49  air_ba937bf13d40fb24  2016-03-12        25     Dining bar   

                   air_area_name   latitude   longitude calendar_date  \
0   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-13   
1   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-14   
2   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-15   
3   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-16   
4   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-18   
5   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-19   
6   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-20   
7   Tōkyō-to Minato-ku Shibakōen  35.658068  139.751599    2016-01-21   
8   Tōkyō-to Minato-ku Shibakōen  35.65